### ISSUE TITLE
Offer standard "All | Unread | Starred" view switcher mode buttons, clean up folder options

### ISSUE THREAD

**Issue by nekohayo** [lwindolf/liferea#811]
Looking at the Liferea preferences dialog tonight, something struck me: in the "Folders" section, nearly all those options are useless:

* _"Show the items of all child feeds when a folder is selected":_ is there ever a case you can imagine someone would want that option to be turned off? If they don't want to see the contents of a folder, don't click on the folder, as simple as that...
* _"Hide read items.":_ useless with the change I'd propose below, but it would depend on making the headerbar UI the default user experience.

The combination of these two options above allow you to view grouped unread items by clicking a folder... but they don't allow you to view the "unread items of an individual feed" for more precision, nor do they allow you to rapidly view "all" (including read) or "starred" (important) items in a folder of feeds (not without going into the preferences dialog, which is cumbersome).

Apps with headerbars not only save space, but also open up new design possibilities and conventions, like the now widespread "[view switcher](https://developer.gnome.org/hig/stable/view-switchers.html.en)" linked buttons design pattern (which can even be seen from the [design patterns front page](https://developer.gnome.org/hig/stable/patterns.html.en)).

So a much more straightforward and flexible UI design to solve the problems I mentioned above would therefore be to **use "view switcher" buttons in the headerbar: _All, Unread, Starred._**
* That way, I could easily decide to show All or Unread items, no matter if I'm in a folder or in an individual feed...
* This would also mean the "Folder" options could be deprecated from the preferences dialog; the remaining "Update all favicons now" button could be moved to the "Feeds" section of the preferences dialog, in the "Feed Cache Handling" subsection.

If you're worried about not having enough space for those three buttons on the headerbar, I would actually recommend removing all the existing buttons on the left except the "Next unread item" one, because I really doubt anyone uses the "<", ">" buttons (since you can click directly on a feed item, or use the keyboard arrow keys) nor the "Mark all as read" button (which I consider dangerous and suggest should be removed in issue #784).

So in my mind, the headerbar would look roughly like this:

`
[ All (%d) | Unread (%d) | Starred ]  (↪️)          Updating feed 45 of 446 — Liferea         (🔍) (☰) (X)
`

...where the two %d are actually indicators of the amount of read/unread items in the current selection, instead of showing that in the titlebar label, but that's an optional change (I just think it makes sense to make it more contextual like that).

As you can see, the changes make so much more sense "as a cohesive whole" that I'd recommend biting the bullet and making the headerbar the default UI, if you dare... even outside GNOME, I don't see the point of a UI not using a headerbar nowadays; it saves so much space and having one unified UI could simplify the code, in theory. I could go more radical than that in my proposed change by replacing the unread count/label by some more useful buttons (like toggling the Unread/starred status on a feed item) but I'm going with the "easy" proposal (if you're interested in a  broader proposal, let me know and I'll be happy to sketch it).

**Comment by rich-coe** [comment]
>> "Show the items of all child feeds when a folder is selected": is there ever a case you can imagine someone would want that option to be turned off? 

You have a folder called 'cat videos' with all the feeds that produce amazing cat information.  The normal behaviour is to show all the items.  It takes a long time to load, and the action for opening a folder is to load all the child items.  Turn it off, and you can navigate the tree folder without a long delay. 

The issue is caused by the slowness in updating the treelist gtk object as the number of objects increases.

**Comment by rich-coe** [comment]
>> "Hide read items.": useless with the change I'd propose below, but it would depend on making the headerbar UI the default user experience.

Reduces the number of items to be shown when clicking a folder to only show the new items.  Again, not useless if your use case is to look at only the unread cat information.

**Comment by nekohayo** [comment]
> [...] the action for opening a folder is to load all the child items. Turn it off, and you can navigate the tree folder without a long delay.

I don't think so, I can fold/unfold a folder by clicking the expander arrow on the left of the folder name, and doing so doesn't trigger opening that folder's view, so there is no performance impact in folding/unfolding folders for me. Or am I misunderstanding what you meant @rich-coe ?
